content/ja/docs/tasks/administer-cluster/configure-upgrade-etcd.md

```diff
@@ -0,0 +1,360 @@
+   1. 各Kubernetes APIサーバーが単一のetcdメンバーと通信している場合、ステップ2で停止したKubernetes APIサーバーを起動します。
+    その後、Kubernetes APIサーバークライアントを再度構成して、停止されたKubernetes APIサーバーへのリクエストをルーティングします。
+    これはロードバランサーを構成することで、多くの場合行われます。
+
+クラスターの再構成に関する詳細については、[etcd再構成ドキュメント](https://etcd.io/docs/current/op-guide/runtime-configuration/#remove-a-member)を参照してください。
+
+## etcdクラスターのバックアップ
+
+すべてのKubernetesオブジェクトはetcdに保存されています。
+定期的にetcdクラスターのデータをバックアップすることは、すべてのコントロールプレーンノードを失うなどの災害シナリオでKubernetesクラスターを復旧するために重要です。
+スナップショットファイルには、すべてのKubernetesの状態と重要な情報が含まれています。
+機密性の高いKubernetesデータを安全に保つために、スナップショットファイルを暗号化してください。
+
+etcdクラスターのバックアップは、etcdのビルトインスナップショットとボリュームスナップショットの2つの方法で実現できます。
+
+### ビルトインスナップショット
+
+etcdはビルトインスナップショットをサポートしています。
+スナップショットは、`etcdctl snapshot save`コマンドを使用してライブメンバーから、あるいはetcdプロセスによって現在使用されていない[データディレクトリ](https://etcd.io/docs/current/op-guide/configuration/#--data-dir)から`member/snap/db`ファイルをコピーして取得できます。
+スナップショットを取得しても、メンバーのパフォーマンスに影響はありません。
+
+以下は、`$ENDPOINT`によって提供されるキースペースのスナップショットを`snapshot.db`ファイルに取る例です:
+
+```shell
+ETCDCTL_API=3 etcdctl --endpoints $ENDPOINT snapshot save snapshot.db
+```
+
+スナップショットを確認します:
+
+```shell
+ETCDCTL_API=3 etcdctl --write-out=table snapshot status snapshot.db
+```
+
+```console
++----------+----------+------------+------------+
+|   HASH   | REVISION | TOTAL KEYS | TOTAL SIZE |
++----------+----------+------------+------------+
+| fe01cf57 |       10 |          7 | 2.1 MB     |
++----------+----------+------------+------------+
+```
+
+### ボリュームスナップショット
+
+etcdがAmazon Elastic Block Storeのようなバックアップをサポートするストレージボリューム上で実行されている場合、ストレージボリュームのスナップショットを取ることによってetcdデータをバックアップします。
+
+### etcdctlオプションを使用したスナップショット
+
+etcdctlによって提供されるさまざまなオプションを使用してスナップショットを取ることもできます。例えば
+
+```shell
+ETCDCTL_API=3 etcdctl -h
+```
+
+はetcdctlから利用可能なさまざまなオプションを一覧表示します。
+例えば、以下のようにエンドポイント、証明書などを指定してスナップショットを取ることができます:
+
+```shell
+ETCDCTL_API=3 etcdctl --endpoints=https://127.0.0.1:2379 \
+  --cacert=<trusted-ca-file> --cert=<cert-file> --key=<key-file> \
+  snapshot save <backup-file-location>
+```
+
+ここで、`trusted-ca-file`、`cert-file`、`key-file`はetcd Podの説明から取得できます。
+
+## etcdクラスターのスケールアウト
+
+etcdクラスターのスケールアウトは、パフォーマンスとのトレードオフで可用性を高めます。
+スケーリングはクラスターのパフォーマンスや能力を高めるものではありません。
+一般的なルールとして、etcdクラスターをスケールアウトまたはスケールインすることはありません。
+etcdクラスターに自動スケーリンググループを設定しないでください。
+公式にサポートされるどんなスケールのプロダクション環境のKubernetesクラスターにおいても、常に静的な5つのメンバーのetcdクラスターを運用することを強く推奨します。
+
+合理的なスケーリングは、より高い信頼性が求められる場合に、3つのメンバーで構成されるクラスターを5つのメンバーにアップグレードすることです。
+既存のクラスターにメンバーを追加する方法については、[etcdの再構成ドキュメント](https://etcd.io/docs/current/op-guide/runtime-configuration/#remove-a-member)を参照してください。
+
+## etcdクラスターの復元
+
+etcdは、[major.minor](http://semver.org/)バージョンのetcdプロセスから取得されたスナップショットからの復元をサポートしています。
+異なるパッチバージョンのetcdからのバージョン復元もサポートされています。
+復元操作は、失敗したクラスターのデータを回復するために用いられます。
+
+復元操作を開始する前に、スナップショットファイルが存在している必要があります。
+これは、以前のバックアップ操作からのスナップショットファイル、または残っている[データディレクトリ](https://etcd.io/docs/current/op-guide/configuration/#--data-dir)からのものである可能性があります。
```

Comment:
ありがとうございます。原文を確認すると、

> Before starting the restore operation, a snapshot file must be present. It can either be a snapshot file from a previous backup operation, or from a remaining [data directory](https://etcd.io/docs/current/op-guide/configuration/#--data-dir).

となっており、 "It can either be ... from <foo> or <bar>" となっていますね。たしかに「○○と○○のどちらのファイルでもよい」の方が適切だと感じました